Zend/zend_smart_str.c

```diff
@@ -169,3 +169,37 @@ ZEND_API void ZEND_FASTCALL _smart_string_alloc(smart_string *str, size_t len)
 		str->c = erealloc2(str->c, str->a + 1, str->len);
 	}
 }
+
+ZEND_API void ZEND_FASTCALL smart_str_append_escaped_truncated(smart_str *str, zend_string *value, size_t length)
+{
+	smart_str_append_escaped(str, ZSTR_VAL(value), MIN(length, ZSTR_LEN(value)));
+
+	if (ZSTR_LEN(value) > length) {
+		smart_str_appendl(str, "...", sizeof("...")-1);
+	}
+}
+
+ZEND_API void ZEND_FASTCALL smart_str_append_scalar(smart_str *dest, zval *value, size_t truncate) {
+	ZEND_ASSERT(Z_TYPE_P(value) <= IS_STRING);
+
+	if (Z_TYPE_P(value) == IS_UNDEF || Z_TYPE_P(value) == IS_NULL) {
+		smart_str_appendl(dest, "NULL", sizeof("NULL")-1);
+	} else if (Z_TYPE_P(value) == IS_TRUE || Z_TYPE_P(value) == IS_FALSE) {
+		smart_str_appends(dest, Z_TYPE_P(value) == IS_TRUE ? "true" : "false");
+	} else if (Z_TYPE_P(value) == IS_DOUBLE) {
+		smart_str dval = {0};
+		smart_str_append_printf(&dval, "%.*H", (int) EG(precision), Z_DVAL_P(value));
```

Comment:
/opt/src/php-src/Zend/zend_smart_str.c: In function ‘smart_str_append_scalar’:
/opt/src/php-src/Zend/zend_smart_str.c:191:38: warning: conversion lacks type at end of format [-Wformat=]
  191 |   smart_str_append_printf(&dval, "%.*H", (int) EG(precision), Z_DVAL_P(value));
      |                                      ^
/opt/src/php-src/Zend/zend_smart_str.c:191:34: warning: too many arguments for format [-Wformat-extra-args]
  191 |   smart_str_append_printf(&dval, "%.*H", (int) EG(precision), Z_DVAL_P(value));

What is this about ?

Reply:
These prints are checked, can't use H